lib/apiservers/service/restapi/handlers/vch_cert_get.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/middleware"
+
+	"github.com/vmware/vic/lib/apiservers/service/models"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/handlers/util"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/operations"
+	"github.com/vmware/vic/lib/config"
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+type VCHCertGet struct{}
+
+func (h *VCHCertGet) Handle(params operations.GetTargetTargetVchVchIDCertificateParams, principal interface{}) middleware.Responder {
+	d, err := buildData(params.HTTPRequest.Context(),
+		url.URL{Host: params.Target},
+		principal.(Credentials).user,
+		principal.(Credentials).pass,
+		params.Thumbprint,
+		nil,
+		nil)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDCertificateDefault(
+			util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	d.ID = params.VchID
+
+	op := trace.NewOperation(params.HTTPRequest.Context(), "vch: %s", params.VchID)
+	c, err := getVCHCert(op, d)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDCertificateDefault(
+			util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	cert := asPemCertificate(c.Cert)
+	return NewGetTargetTargetVchVchIDCertificateOK(cert.Pem)
+}
+
+func getVCHCert(op trace.Operation, d *data.Data) (*config.RawCertificate, error) {
+	vchConfig, err := getVCHConfig(op, d)
+	if err != nil {
+		return nil, err
+	}
+
+	if vchConfig.HostCertificate.IsNil() {
+		return nil, fmt.Errorf("No certificate found for VCH %s", d.ID)
```

Comment:
I think you'll want to use `util.NewError` or `util.WrapError` here so that you can specify a status code. It'll default to 500, but 404 is probably more appropriate in this case (since the resource being requested _is_ the certificate we weren't able to find).